### DISCUSSION TITLE
Install R packages not working

### DISCUSSION THREAD

**Discussion by gitmiguelribeiro** [posit-dev/positron#5355] [Q&A] [answered]
I'm using R version 4.4.2 (2024-10-31 ucrt) -- "Pile of Leaves", and it works fine in Rstudio.

For some reason, I'm not able to install.packages or call them. I keep getting the error below:
> install.packages('tidyverse')
Warning: unable to access index for repository [https://cran.rstudio.com/src/contrib:](vscode-file://vscode-app/c:/Program%20Files/Positron/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html#)
  cannot open URL '[https://cran.rstudio.com/src/contrib/PACKAGES](vscode-file://vscode-app/c:/Program%20Files/Positron/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html#)'
Warning: unable to access index for repository [https://cran.rstudio.com/bin/windows/contrib/4.4:](vscode-file://vscode-app/c:/Program%20Files/Positron/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html#)
  cannot open URL '[https://cran.rstudio.com/bin/windows/contrib/4.4/PACKAGES](vscode-file://vscode-app/c:/Program%20Files/Positron/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html#)'
Warning message:
package ‘tidyverse’ is not available for this version of R

A version of this package for your version of R might be available elsewhere,
see the ideas at
[https://cran.r-project.org/doc/manuals/r-patched/R-admin.html#Installing-packages](vscode-file://vscode-app/c:/Program%20Files/Positron/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html#)

**Comment by jennybc** [comment]
What happens if you try to install tidyverse from _outside_ of Positron? The above looks more like a ?hopefully temporary? glitch in the CRAN mirror than anything to do with Positron.

**Reply by gitmiguelribeiro** [reply]
Works normally. Using Rstudio I can install and load them.

**Reply by jennybc** [reply]
What do you get for `getOption("repos")` in the 2 scenarios (working, not working)?

**Reply by gitmiguelribeiro** [reply]
Not working scenario (Positron): 

CRAN 
"[https://cran.rstudio.com/](vscode-file://vscode-app/c:/Program%20Files/Positron/resources/app/out/vs/code/electron-sandbox/workbench/workbench.html#)" 
attr(,"IDE")
[1] TRUE


Working scenario (RStudio):
                      CRAN 
"http://cran.rstudio.com/" 
attr(,"RStudio")
[1] TRUE

**Reply by jennybc** [reply, marked as answer]
Is there any more context to this? For example, is renv or another environment management tool involved? Do you have an `.Rprofile`?

**Reply by jennybc** [reply]
One other line of inquiry is whether you're behind some sort of firewall or proxy and things are configured differently w.r.t. RStudio vs. Positron.

**Reply by gitmiguelribeiro** [reply]
Not really. I just installed Positron and it doesn't install any packages. Rstudio on the other hand install them normally, but I had to uncheck the Use secure download method for HTTP in order for it to work.
I have an .Rprofile with the following line only:
.libPaths("C:/Users/141640/AppData/Local/R/Rpackages")

**Reply by gitmiguelribeiro** [reply]
Found the solution in https://support.posit.co/hc/en-us/articles/206827897-Secure-Package-Downloads-for-R

I added the lines bellow in the .Rprofile and also run the in Positron console. Works fine now, thank you for your questions, they helped finding the answer.

options(download.file.method = "wininet")
options(repos = "https://cran.rstudio.com")

**Reply by jennybc** [reply]
Glad you're unstuck! I will emphasize that you want to be using https here (vs. disabling secure download). So your experience in RStudio raises an eyebrow.

I still find it rather peculiar that you needed to configure options in Positron 🤔

**Reply by juliasilge** [reply]
I wrote up https://github.com/posit-dev/positron/issues/5390 to try to capture the situation as best as I understand it.

**Reply by gitmiguelribeiro** [reply]
I'm using indeed a corporate network. Also my work peers have the same issue, which is resolved by disabling the secure download.
